tests/filecheck/backend/rvscf_lowering_labels.mlir

```diff
@@ -0,0 +1,112 @@
+// RUN: xdsl-opt -p lower-riscv-scf-to-labels --split-input-file %s | filecheck %s
+
+builtin.module {
+    "riscv_func.func"() ({
+    ^0(%0 : !riscv.reg<a0>, %1 : !riscv.reg<a1>):
+        %2 = riscv.mv %0 : (!riscv.reg<a0>) -> !riscv.reg<a0>
+        %3 = riscv.mv %1 : (!riscv.reg<a1>) -> !riscv.reg<a1>
```

Comment:
I am surprised that the input test case uses `riscv.mv` instructions? I feel these just are not needed and could be removed to reduce noise? Or do I miss sth here?

Reply:
This is an artefact of the register allocation approach. The idea is that the calling convention requires the funciton arguments to be in `a{k}` but we might want to reuse those registers inside the function. So the first thing we do is move them to a new register, which might end up being the same one, in which case the move will be eliminated during canonicalization.

Reply:
I feel it would be useful to drop this leftover artefacts from register allocation in this test case. This will make the test easier to understand.

Reply:
https://github.com/xdslproject/xdsl/pull/1456